### ISSUE TITLE
Fails to install due to no RobustCloudCommand

### ISSUE THREAD

**Issue by CTI-Corey** [T0pCyber/hawk#117]
It appears that RobustCloudCommand was de-listed?
https://www.powershellgallery.com/packages/RobustCloudCommand/ 

It can still be installed manually with 

`Install-Module -Name RobustCloudCommand -RequiredVersion 2.1.0`

However the Install-Module -Name Hawk fails currently due to this.

**Comment by jonnybottles** [comment]
https://learn.microsoft.com/en-us/compliance/assurance/assurance-resource-limits

**Comment by jonnybottles** [comment]
# BLUF (Bottom Line Up Front)
Build failures caused by the missing `RobustCloudCommand` dependency have been resolved by removing it from the manifest. Since its only usage was in the now-inactive `Get-HawkTenantInboxRules` function, this removal fixes builds while maintaining all current functionality. A separate issue has been opened to track the reimplementation of the tenant-wide inbox rule scanning capability.

---

## Key Actions
- Removed `RobustCloudCommand` from manifest dependencies.
- Opened a[ new issue](https://github.com/T0pCyber/hawk/issues/156) to reimplement tenant-wide inbox rule retrieval.

These actions unblock current development while ensuring the security scanning capability is properly tracked for future reimplementation.

---

## Background
Build failures were occurring due to a dependency on the `RobustCloudCommand` module, which is no longer available in the PowerShell Gallery.

### Investigation Findings

#### Dependency Analysis
- `RobustCloudCommand` was listed as a dependency in the manifest.
- Its only usage was in `Get-HawkTenantInboxRules`.
- `Get-HawkTenantInboxRules` was not called by any active functions, including `Start-HawkTenantInvestigation`.

#### Function Purpose
- `Get-HawkTenantInboxRules` was designed to scan all tenant mailboxes for:
  - Malicious inbox rules (via `Get-HawkUserInboxRule`).
  - Unauthorized email forwarding (via `Get-HawkUserEmailForwarding`).
- It relied on `RobustCloudCommand` to handle throttling when scanning large mailbox sets.

#### Changelog Review
- Version 2.0.2 updated `Get-HawkTenantInboxRules` to support `RobustCloudCommand 2.0.1`.
- No subsequent changelog entries indicated removal or deprecation of the function.
- The function was disconnected from the main workflow but remained in the codebase.

---

## Current State
The `RobustCloudCommand` dependency has been removed from the manifest since the only dependent function was not actively used. This resolves the immediate build failure.

---

## Way Forward
A follow-up issue has been created to address related tasks:

### Reimplement Tenant-Wide Inbox Rule Scanning
#### If using RobustCloudCommand:
- Make assessment of it works in the current project
- If so, implement it

#### If not using RobustCloudCommand:

- Replace calls to `RobustCloudCommand` with:
  - PowerShell's built-in `Start-Job` for parallel processing.
  - Use Microsoft Graph API for mailbox enumeration and rule retrieval.
- Update the `Hawk` module manifest to remove `RobustCloudCommand` from dependencies.
- Add new PowerShell cmdlets for mailbox batching to prevent throttling issues.
